html5lib is not available on Solaris, so let's skip tests depending on it.

GitHub changed its frontend to be entirely client-side rendered, meaning that
anchors are no longer part of the downloaded page.

Not intended for upstream.

--- Sphinx-6.1.3/tests/test_smartquotes.py
+++ Sphinx-6.1.3/tests/test_smartquotes.py
@@ -1,7 +1,6 @@
 """Test smart quotes."""
 
 import pytest
-from html5lib import HTMLParser
 
 
 @pytest.mark.sphinx(buildername='html', testroot='smartquotes', freshenv=True)
@@ -12,6 +11,7 @@ def test_basic(app, status, warning):
     assert '<p>– “Sphinx” is a tool that makes it easy …</p>' in content
 
 
+@pytest.mark.skip(reason="html5lib.HTMLParser is not available on Solaris")
 @pytest.mark.sphinx(buildername='html', testroot='smartquotes', freshenv=True)
 def test_literals(app, status, warning):
     app.build()
--- Sphinx-6.1.3/tests/test_domain_std.py
+++ Sphinx-6.1.3/tests/test_domain_std.py
@@ -5,7 +5,6 @@ from unittest import mock
 import pytest
 from docutils import nodes
 from docutils.nodes import definition, definition_list, definition_list_item, term
-from html5lib import HTMLParser
 
 from sphinx import addnodes
 from sphinx.addnodes import (desc, desc_addname, desc_content, desc_name, desc_signature,
@@ -358,6 +357,7 @@ def test_multiple_cmdoptions(app):
     assert domain.progoptions[('cmd', '--output')] == ('index', 'cmdoption-cmd-o')
 
 
+@pytest.mark.skip(reason="html5lib.HTMLParser is not available on Solaris")
 @pytest.mark.sphinx(testroot='productionlist')
 def test_productionlist(app, status, warning):
     app.builder.build_all()
--- Sphinx-6.1.3/tests/test_build_html.py
+++ Sphinx-6.1.3/tests/test_build_html.py
@@ -7,7 +7,7 @@ from pathlib import Path
 from unittest.mock import ANY, call, patch
 
 import pytest
-from html5lib import HTMLParser
+pytestmark = pytest.mark.skip("html5lib.HTMLParser is not available on Solaris")
 
 from sphinx.builders.html import validate_html_extra_path, validate_html_static_path
 from sphinx.errors import ConfigError
--- Sphinx-6.1.3/tests/roots/test-linkcheck/links.txt
+++ Sphinx-6.1.3/tests/roots/test-linkcheck/links.txt
@@ -13,7 +13,7 @@ Some additional anchors to exercise igno
 * `Complete nonsense <https://localhost:7777/doesnotexist>`_
 * `Example valid local file <conf.py>`_
 * `Example invalid local file <path/to/notfound>`_
-* https://github.com/sphinx-doc/sphinx/blob/master/sphinx/__init__.py#L2
+* https://www.sphinx-doc.org/en/master/#get-started
 
 .. image:: https://www.google.com/image.png
 .. figure:: https://www.google.com/image2.png
